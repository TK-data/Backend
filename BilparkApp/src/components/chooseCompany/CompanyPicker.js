import React from 'react';
import { StyleSheet, View, Text } from 'react-native';
import { Button, Picker } from 'native-base';
import { connect } from 'react-redux';

// import action to get companies

const style = StyleSheet.create({
})

const CompanyPicker = ({ getCompanies, postCompany, companies }) => {

  return (
    <View>
      <View>
        <Picker />
      </View>
      <View>
        <Button light>
          <Text>
            Send
          </Text>
        </Button>
      </View>
    </View>
  );
}

const mapDispatchToProps = (dispatch) => {
  return {
    getCompanies: () => null,
    postCompany: () => null,
  };
};

const mapStateToProps = (state) => {
  return {
    companies: state.companies,
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(CompanyPicker);
