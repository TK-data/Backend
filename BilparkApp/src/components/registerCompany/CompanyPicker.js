import React from 'react';
import { StyleSheet, Dimensions, TouchableOpacity, Platform } from 'react-native';
import { Picker, Spinner, View, Text } from 'native-base';
import { connect } from 'react-redux';

import { saveCompany, selectCompany } from '../../actions/registerCompany';

const width = Dimensions.get('window').width;

const styles = StyleSheet.create({
  container: {
    width: width / 2,
    marginTop: 20,
  },
  button: {
    alignSelf: 'stretch',
  },
  text: {
    textAlign: 'center',
    color: 'white',
    fontSize: 20,
    marginTop: 10,
  },
  picker: {
    backgroundColor: 'white',
  },
});

const CompanyPicker = ({ hasErrored, isLoading, selectedCompany, changeSelect, postCompany, companies }) => {

  if (hasErrored) {
    return (
      <View style={styles.container}>
        <Text style={styles.text}>Noe gikk galt n√•r firmaer skulle hentes..</Text>
      </View>
    );
  }

  if (isLoading) {
    return (
      <View>
        <Spinner color="white" />
      </View>
    );
  }

  if (Platform.OS === 'android') {
    return (
      <View style={styles.container}>
        <View>
          <Picker
            iosHeader="Velg Selskap"
            mode="dropdown"
            style={styles.picker}
            selectedValue={selectedCompany}
            onValueChange={value => changeSelect(value)}
            placeholder="Velg selskap"
          >
            { Platform.OS === 'android' && <Picker.Item label="Velg" value="" /> }
            {companies.map(company =>
              (<Picker.Item
                key={company.CompanyID}
                label={company.CompanyName}
                value={company.CompanyID}
              />))}
          </Picker>
        </View>
        <View>
          <TouchableOpacity
            onPress={() => ((selectedCompany === '') ? null : postCompany(selectedCompany))}
          >
            <Text style={styles.text}>
              Send
            </Text>
          </TouchableOpacity>
        </View>
      </View>
    );
  }

  // if ios
  return (
    <View style={styles.container}>
      <View>
        <Picker
          iosHeader="Velg Selskap"
          mode="dropdown"
          style={styles.picker}
          selectedValue={selectedCompany}
          onValueChange={value => changeSelect(value)}
          placeholder="Velg selskap"
        >
          {companies.map(company =>
            (<Picker.Item
              key={company.CompanyID}
              label={company.CompanyName}
              value={company.CompanyID}
            />))}
        </Picker>
      </View>
      <View>
        <TouchableOpacity
          onPress={() => ((selectedCompany === '') ? null : postCompany(selectedCompany))}
        >
          <Text style={styles.text}>
            Send
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const mapDispatchToProps = (dispatch) => {
  return {
    postCompany: value => dispatch(saveCompany(value)),
    changeSelect: value => dispatch(selectCompany(value)),
  };
};

const mapStateToProps = (state) => {
  return {
    companies: state.registerCompany.companies,
    hasErrored: state.registerCompany.hasErrored,
    isLoading: state.registerCompany.isLoading,
    selectedCompany: state.registerCompany.selectedCompany,
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(CompanyPicker);
