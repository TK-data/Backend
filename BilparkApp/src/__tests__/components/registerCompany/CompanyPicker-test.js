import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import CompanyPicker from '../../../components/registerCompany/CompanyPicker';

const initialState = {
  registerCompany: {
    companies: [{ CompanyID: 5, CompanyName: 'Bedrift1' }, { CompanyID: 8, CompanyName: 'Bedrift2' }],
    selectedCompany: 5,
  },
};

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Testing FuelRefillScreen', () => {
  it('Renders as expected', () => {
    const comp = shallow(
      <CompanyPicker />,
      { context: { store: mockStore(initialState) } },
    );
    expect(comp.dive()).toMatchSnapshot();
  });
});
