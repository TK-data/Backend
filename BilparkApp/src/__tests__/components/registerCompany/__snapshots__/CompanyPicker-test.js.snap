// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CompanyPicker Renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CompanyPicker
    changeSelect={[Function]}
    companies={
      Array [
        Object {
          "CompanyID": 5,
          "CompanyName": "Bedrift1",
        },
        Object {
          "CompanyID": 8,
          "CompanyName": "Bedrift2",
        },
      ]
    }
    hasErrored={undefined}
    isLoading={undefined}
    postCompany={[Function]}
    selectedCompany={5}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(ViewNB)
          virtual={undefined}
        >
          <Styled(PickerNB)
            headerBackButtonText="<"
            iosHeader="Velg Selskap"
            mode="dropdown"
            onValueChange={[Function]}
            placeholder="Velg selskap"
            selectedValue={5}
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "white",
              }
            }
            virtual={undefined}
          >
            <Item
              label="Bedrift1"
              value={5}
            />
            <Item
              label="Bedrift2"
              value={8}
            />
          </Styled(PickerNB)>
        </Styled(ViewNB)>,
        <Styled(ViewNB)
          virtual={undefined}
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
          >
            <Styled(Text)
              style={
                Object {
                  "color": "white",
                  "fontSize": 20,
                  "marginTop": 10,
                  "textAlign": "center",
                }
              }
              virtual={undefined}
            >
              Lagre
            </Styled(Text)>
          </TouchableOpacity>
        </Styled(ViewNB)>,
      ],
      "style": Object {
        "marginTop": 20,
        "width": 375,
      },
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Styled(PickerNB)
            headerBackButtonText="<"
            iosHeader="Velg Selskap"
            mode="dropdown"
            onValueChange={[Function]}
            placeholder="Velg selskap"
            selectedValue={5}
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "white",
              }
            }
            virtual={undefined}
          >
            <Item
              label="Bedrift1"
              value={5}
            />
            <Item
              label="Bedrift2"
              value={8}
            />
          </Styled(PickerNB)>,
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Item
                label="Bedrift1"
                value={5}
              />,
              <Item
                label="Bedrift2"
                value={8}
              />,
            ],
            "headerBackButtonText": "<",
            "iosHeader": "Velg Selskap",
            "mode": "dropdown",
            "onValueChange": [Function],
            "placeholder": "Velg selskap",
            "selectedValue": 5,
            "style": Object {
              "alignSelf": "center",
              "backgroundColor": "white",
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "5",
              "nodeType": "class",
              "props": Object {
                "label": "Bedrift1",
                "value": 5,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "8",
              "nodeType": "class",
              "props": Object {
                "label": "Bedrift2",
                "value": 8,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
          >
            <Styled(Text)
              style={
                Object {
                  "color": "white",
                  "fontSize": 20,
                  "marginTop": 10,
                  "textAlign": "center",
                }
              }
              virtual={undefined}
            >
              Lagre
            </Styled(Text)>
          </TouchableOpacity>,
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": <Styled(Text)
              style={
                Object {
                  "color": "white",
                  "fontSize": 20,
                  "marginTop": 10,
                  "textAlign": "center",
                }
              }
              virtual={undefined}
            >
              Lagre
            </Styled(Text)>,
            "onPress": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Lagre",
              "style": Object {
                "color": "white",
                "fontSize": 20,
                "marginTop": 10,
                "textAlign": "center",
              },
              "virtual": undefined,
            },
            "ref": null,
            "rendered": "Lagre",
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(ViewNB)
            virtual={undefined}
          >
            <Styled(PickerNB)
              headerBackButtonText="<"
              iosHeader="Velg Selskap"
              mode="dropdown"
              onValueChange={[Function]}
              placeholder="Velg selskap"
              selectedValue={5}
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "white",
                }
              }
              virtual={undefined}
            >
              <Item
                label="Bedrift1"
                value={5}
              />
              <Item
                label="Bedrift2"
                value={8}
              />
            </Styled(PickerNB)>
          </Styled(ViewNB)>,
          <Styled(ViewNB)
            virtual={undefined}
          >
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
            >
              <Styled(Text)
                style={
                  Object {
                    "color": "white",
                    "fontSize": 20,
                    "marginTop": 10,
                    "textAlign": "center",
                  }
                }
                virtual={undefined}
              >
                Lagre
              </Styled(Text)>
            </TouchableOpacity>
          </Styled(ViewNB)>,
        ],
        "style": Object {
          "marginTop": 20,
          "width": 375,
        },
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(PickerNB)
              headerBackButtonText="<"
              iosHeader="Velg Selskap"
              mode="dropdown"
              onValueChange={[Function]}
              placeholder="Velg selskap"
              selectedValue={5}
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "white",
                }
              }
              virtual={undefined}
            >
              <Item
                label="Bedrift1"
                value={5}
              />
              <Item
                label="Bedrift2"
                value={8}
              />
            </Styled(PickerNB)>,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Item
                  label="Bedrift1"
                  value={5}
                />,
                <Item
                  label="Bedrift2"
                  value={8}
                />,
              ],
              "headerBackButtonText": "<",
              "iosHeader": "Velg Selskap",
              "mode": "dropdown",
              "onValueChange": [Function],
              "placeholder": "Velg selskap",
              "selectedValue": 5,
              "style": Object {
                "alignSelf": "center",
                "backgroundColor": "white",
              },
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "5",
                "nodeType": "class",
                "props": Object {
                  "label": "Bedrift1",
                  "value": 5,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "8",
                "nodeType": "class",
                "props": Object {
                  "label": "Bedrift2",
                  "value": 8,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
            >
              <Styled(Text)
                style={
                  Object {
                    "color": "white",
                    "fontSize": 20,
                    "marginTop": 10,
                    "textAlign": "center",
                  }
                }
                virtual={undefined}
              >
                Lagre
              </Styled(Text)>
            </TouchableOpacity>,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeOpacity": 0.2,
              "children": <Styled(Text)
                style={
                  Object {
                    "color": "white",
                    "fontSize": 20,
                    "marginTop": 10,
                    "textAlign": "center",
                  }
                }
                virtual={undefined}
              >
                Lagre
              </Styled(Text)>,
              "onPress": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Lagre",
                "style": Object {
                  "color": "white",
                  "fontSize": 20,
                  "marginTop": 10,
                  "textAlign": "center",
                },
                "virtual": undefined,
              },
              "ref": null,
              "rendered": "Lagre",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
